website/www/site/content/en/documentation/sdks/python-machine-learning.md

```diff
@@ -215,6 +215,53 @@ with pipeline as p:
 
 If you are unsure if your data is keyed, you can also use `MaybeKeyedModelHandler`.
 
+You can also use a `KeyedModelHandler` to load several different models based on their associated key:
+
+```
+from apache_beam.ml.inference.base import KeyedModelHandler
+keyed_model_handler = KeyedModelHandler([
+  KeyModelMapping(['key1'], PytorchModelHandlerTensor(<config1>)),
+  KeyModelMapping(['key2', 'key3'], PytorchModelHandlerTensor(<config2>))
+])
+with pipeline as p:
+   data = p | beam.Create([
+      ('key1', torch.tensor([[1,2,3],[4,5,6],...])),
+      ('key2', torch.tensor([[1,2,3],[4,5,6],...])),
+      ('key3', torch.tensor([[1,2,3],[4,5,6],...])),
+   ])
+   predictions = data | RunInference(keyed_model_handler)
+```
+
+The previous example will load a model using `config1` and use that for inference for all examples associated
+with `key1`, and will load a model using `config2` and use that for all examples associated with `key2` and `key3`.
+
+There are memory risks associated with loading multiple models at once. By default, `KeyedModelHandler` will not
+limit the number of models loaded into memory at once. This means that if not all models fit into memory at once,
+your pipeline will likely fail with an Out of Memory exception. To avoid this, you can provide a hint about the
+maximum number of models loaded at once.=:
+
+```
+mhs = [
+  KeyModelMapping(['key1'], PytorchModelHandlerTensor(<config1>)),
+  KeyModelMapping(['key2', 'key3'], PytorchModelHandlerTensor(<config2>)),
+  KeyModelMapping(['key4'], PytorchModelHandlerTensor(<config3>)),
+  KeyModelMapping(['key5', 'key6', 'key7'], PytorchModelHandlerTensor(<config4>)),
+]
+keyed_model_handler = KeyedModelHandler(mhs, max_models_per_worker_hint=2)
+```
+
+The previous example will load at most 2 models per worker at any given time, and will unload models that aren't
```

Comment:
> load at most 2 models per worker 

Users might perceive it as a guarantee, and come to us if/when they see a single OOM error.  If this cannot be guaranteed, we can phrase that the upper  ceiling is enforced as best effort. Or mention that there may be some delay between when the model is unloaded and the memory is freed.

Reply:
I mentioned that there's some delay, let me know if you think the wording is ok